ndk/src/media/error.rs

```diff
@@ -4,77 +4,77 @@ use thiserror::Error;
     ErrorResult(MediaErrorResult),
-    #[error("unknown Media error result ({0})")]
-    UnknownResult(i32),
+    #[error("unknown Media error result ({0:?})")]
+    UnknownResult(ffi::media_status_t),
 }
 
 impl NdkMediaError {
     pub(crate) fn from_status(status: ffi::media_status_t) -> Result<()> {
         use MediaErrorResult::*;
         let result = match status {
-            ffi::media_status_t_AMEDIA_OK => return Ok(()),
-            ffi::media_status_t_AMEDIACODEC_ERROR_INSUFFICIENT_RESOURCE => {
+            ffi::media_status_t::AMEDIA_OK => return Ok(()),
+            ffi::media_status_t::AMEDIACODEC_ERROR_INSUFFICIENT_RESOURCE => {
                 CodecErrorInsufficientResource
             }
-            ffi::media_status_t_AMEDIACODEC_ERROR_RECLAIMED => CodecErrorReclaimed,
-            ffi::media_status_t_AMEDIA_ERROR_UNKNOWN => ErrorUnknown,
-            ffi::media_status_t_AMEDIA_ERROR_MALFORMED => ErrorMalformed,
-            ffi::media_status_t_AMEDIA_ERROR_UNSUPPORTED => ErrorUnsupported,
-            ffi::media_status_t_AMEDIA_ERROR_INVALID_OBJECT => ErrorInvalidObject,
-            ffi::media_status_t_AMEDIA_ERROR_INVALID_PARAMETER => ErrorInvalidParameter,
-            ffi::media_status_t_AMEDIA_ERROR_INVALID_OPERATION => ErrorInvalidOperation,
-            ffi::media_status_t_AMEDIA_ERROR_END_OF_STREAM => ErrorEndOfStream,
-            ffi::media_status_t_AMEDIA_ERROR_IO => ErrorIo,
-            ffi::media_status_t_AMEDIA_ERROR_WOULD_BLOCK => ErrorWouldBlock,
-            ffi::media_status_t_AMEDIA_DRM_ERROR_BASE => DrmErrorBase,
-            ffi::media_status_t_AMEDIA_DRM_NOT_PROVISIONED => DrmNotProvisioned,
-            ffi::media_status_t_AMEDIA_DRM_RESOURCE_BUSY => DrmResourceBusy,
-            ffi::media_status_t_AMEDIA_DRM_DEVICE_REVOKED => DrmDeviceRevoked,
-            ffi::media_status_t_AMEDIA_DRM_SHORT_BUFFER => DrmShortBuffer,
-            ffi::media_status_t_AMEDIA_DRM_SESSION_NOT_OPENED => DrmSessionNotOpened,
-            ffi::media_status_t_AMEDIA_DRM_TAMPER_DETECTED => DrmTamperDetected,
-            ffi::media_status_t_AMEDIA_DRM_VERIFY_FAILED => DrmVerifyFailed,
-            ffi::media_status_t_AMEDIA_DRM_NEED_KEY => DrmNeedKey,
-            ffi::media_status_t_AMEDIA_DRM_LICENSE_EXPIRED => DrmLicenseExpired,
-            ffi::media_status_t_AMEDIA_IMGREADER_ERROR_BASE => ImgreaderErrorBase,
-            ffi::media_status_t_AMEDIA_IMGREADER_NO_BUFFER_AVAILABLE => ImgreaderNoBufferAvailable,
-            ffi::media_status_t_AMEDIA_IMGREADER_MAX_IMAGES_ACQUIRED => ImgreaderMaxImagesAcquired,
-            ffi::media_status_t_AMEDIA_IMGREADER_CANNOT_LOCK_IMAGE => ImgreaderCannotLockImage,
-            ffi::media_status_t_AMEDIA_IMGREADER_CANNOT_UNLOCK_IMAGE => ImgreaderCannotUnlockImage,
-            ffi::media_status_t_AMEDIA_IMGREADER_IMAGE_NOT_LOCKED => ImgreaderImageNotLocked,
+            ffi::media_status_t::AMEDIACODEC_ERROR_RECLAIMED => CodecErrorReclaimed,
```

Comment:
All these conversions are a bit odd too, `MediaErrorResult` appears to be `repr(i32)` for a reason though we don't use it to our advantage. Yes, casting unknown values is UB so perhaps we should just use `ffi::media_status_t` directly as suggested above?